### DISCUSSION TITLE
Bucket4j is blocking websocket connection.

### DISCUSSION THREAD

**Discussion by ss-guvenc** [vladimir-bukhtoyarov/bucket4j#252] [Q&A]
Bucket4j Rate limiter is blocking websocket connection. 

After i enable rate limiter on rest endpoint server, WebSocket Client with SockJS can not establish connection with server.

Connection status remains connected on client site, after a certain time, the connection is closed.

Is There any configuration or solution for fixing this problem.

**Comment by vladimir-bukhtoyarov** [comment]
Hello @ss-guvenc 

The Bucket4j does not provide integration with any WebSocket library, so there can not be any configuration option. 

There is only point that I advice to check **If you use any distributed bucket4j integration**(for example bucket4j-hazelcast) and in same time Websocket framework that you use, **requires to write non-blocking code** then obviously you need to switch to [asynchrounous Bucket4j API](https://bucket4j.com/7.4.0/toc.html#asynchronous-api).
